lib/OpenCloud/ObjectStore/Service.php

```diff
@@ -67,6 +67,18 @@ public function getCdnService()
     }
 
     /**
+     * List all available containers.
+     *
+     * @param array $filter
+     * @return \OpenCloud\Common\Collection\PaginatedIterator
+     */
+    public function listContainers(array $filter = array())
```

Comment:
Thank you for refactoring this down. It was silly to have a base class with conditional logic depending on the type of the instance.